.github/workflows/build-and-test.yml

```diff
@@ -34,6 +34,35 @@ jobs:
     - name: Build all classes
       run: ./gradlew --stacktrace classes
 
+    - name: Install Node
```

Comment:
Can you move the new steps to a separate job, e.g. `build-ui`? This provides faster feedback as the builds run in parallel and you can see immediately in a PR if the Gradle or the PNPM builds failed.

Reply:
Done.